test/unittests/config_test.js

```diff
@@ -0,0 +1,98 @@
+import UserConfig from '../../resources/user_config.js';
+import chai from 'chai';
+
+const { assert } = chai;
+
+const tests = {
+  sortUserFiles: () => {
+    const unsortedKeys = [
+      'b but with some spaces/file.js',
+      'b/subdir1/z/z/z/nested_file.js',
+      'a/some.js',
+      'root_file1.js',
+      'zzz/sleepy.js',
+      'B/subDir2/second.js',
+      'b/some_file.js',
+      'b/subdir1/file.js',
+      'root_file2.js',
+      'a/some file with spaces oof why.js',
+      'b/subdir2/first.js',
+    ];
+    const sortedKeys = [
+      'a/some file with spaces oof why.js',
+      'a/some.js',
```

Comment:
I'm not sure what I think about this.  This happens because `some ` sorts before `some.`, but for folders, `b` sorts before `b but with some spaces`.

Should we sort filenames ignoring extensions? Or is sorting the full filename ok? What do you think about this inconsistency (or maybe it isn't one).

Reply:
Hmmm, interesting. I guess my thought as a human with context is that a filename is different from an extension and that on Windows, I might not even have visible file extensions enabled. I would say that the most correct thing to do would be to sort without file extensions and then put them back, but that sounds like a lot of effort for something that I think everyone universally agrees is bad. I would argue that having the test is enough such that it can be pointed to if someone _really_ has an issue, but I think we're at like a corner of edge cases from both Windows and cactbot points of view, and there's a debug output that can be used as a frontline defense.

That being said, if you want to tackle it, go for it; I just don't think it's necessary.

Reply:
Done!